lib/ecto/type.ex

```diff
@@ -224,6 +224,67 @@ defmodule Ecto.Type do
     end
   end
 
+  @doc false
+  def __typespec__(:id) do
+    quote do: integer()
+  end
+  def __typespec__(:binary_id) do
+    quote do: binary()
+  end
+  def __typespec__(:integer) do
+    quote do: integer()
+  end
+  def __typespec__(:float) do
+    quote do: float()
+  end
+  def __typespec__(:boolean) do
+    quote do: boolean()
+  end
+  def __typespec__(:string) do
+    quote do: String.t()
+  end
+  def __typespec__(:binary) do
+    quote do: binary()
+  end
+  def __typespec__({:array, inner}) do
+    quote do: [Macro.escape(Ecto.Type.__typespec__(unquote(inner)))]
+  end
+  def __typespec__(:map) do
+    quote do: map()
+  end
+  def __typespec__({:map, inner}) do
+    quote do: [Macro.escape(Ecto.Type.__typespec__(unquote(inner)))]
```

Comment:
This should be 
```
quote do: %{String.t => Macro.escape(Ecto.Type.__typespec__(unquote(inner)))}
```

Reply:
Good call. Macro.escape cannot be used here like that because:

```elixir
iex(1)> t Friends.Person
@type t() :: %Friends.Person{
        __meta__: term(),
        age: integer(),
        first_name: String.t(),
        id: integer(),
        last_name: String.t(),
        map: [Macro.escape(Ecto.Type.__typespec__(:integer))]
      }
```

I'll investigate and add unit test. Looks like it wasn't quite ready after all, sorry for noise.